### PR TITLE
[PR check] Update Helm Repo to redhat-developer/rhdh-chart

### PR DESCRIPTION
**Pull request by subhashkhileri**
## Description

[PR check]Update Helm Repository from 'janus-idp/helm-backstage' to 'redhat-developer/rhdh-chart'
[e2e tests] Implement Enhanced Backstage Service Availability Check in Deployment Script

## Which issue(s) does this PR fix
https://issues.redhat.com/browse/RHIDP-1194
https://issues.redhat.com/browse/RHIDP-1213

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: .ibm/pipelines/openshift-tests.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ trap cleanup EXIT
 add_helm_repos() {
   helm version
 
-  declare -a repos=("bitnami=https://charts.bitnami.com/bitnami" "backstage=https://backstage.github.io/charts" "janus-idp=https://janus-idp.github.io/helm-backstage" "${HELM_REPO_NAME}=${HELM_REPO_URL}")
+  declare -a repos=("bitnami=https://charts.bitnami.com/bitnami" "backstage=https://backstage.github.io/charts" "rhdh-chart=https://redhat-developer.github.io/rhdh-chart" "${HELM_REPO_NAME}=${HELM_REPO_URL}")
```
Comment (josephca) [inline .ibm/pipelines/openshift-tests.sh:22]: As we use 'rhdh-chart', we may omit to add redundant repository, "${HELM_REPO_NAME}=${HELM_REPO_URL}". Also, we may remove the environment property from the pipeline.

**Review by josephca — APPROVED** [review]
/lgtm
/approve

File: .ibm/pipelines/openshift-tests.sh (hunk 1 above)
Comment (nickboldt) [reply .ibm/pipelines/openshift-tests.sh]: note that the readme at that URL is still in progress... https://github.com/redhat-developer/rhdh-chart/pull/4

File: .ibm/pipelines/openshift-tests.sh [hunk 2]
```diff
@@ -196,7 +196,7 @@ main() {
 
   echo "Tag name with short SHA: pr-${GIT_PR_NUMBER}-${SHORT_SHA}"
 
-  helm upgrade -i ${RELEASE_NAME} -n ${NAME_SPACE} ${HELM_REPO_NAME}/${HELM_IMAGE_NAME} --version ${CHART_VERSION} -f $DIR/value_files/${HELM_CHART_VALUE_FILE_NAME} --set global.clusterRouterBase=${K8S_CLUSTER_ROUTER_BASE} --set upstream.backstage.image.tag=pr-${GIT_PR_NUMBER}-${SHORT_SHA}
+  helm upgrade -i ${RELEASE_NAME} -n ${NAME_SPACE} rhdh-chart/redhat-developer-hub --version ${CHART_VERSION} -f $DIR/value_files/${HELM_CHART_VALUE_FILE_NAME} --set global.clusterRouterBase=${K8S_CLUSTER_ROUTER_BASE} --set upstream.backstage.image.tag=pr-${GIT_PR_NUMBER}-${SHORT_SHA}
```
Comment (nickboldt) [inline .ibm/pipelines/openshift-tests.sh:199]: are we sure this is correct? 

looking at https://artifacthub.io/packages/helm/rhdh-chart/backstage I see the install instructions are:

```
helm install my-backstage oci://ghcr.io/redhat-developer/rhdh-chart/backstage --version 2.13.2
```

File: .ibm/pipelines/openshift-tests.sh [hunk 3]
```diff
@@ -153,13 +153,34 @@ run_tests() {
...
+  local wait_seconds=30
+
+  echo "Checking if Backstage is up and running at $url"
+
+  for ((i=1; i<=max_attempts; i++)); do
+    # Get the status code
+    local http_status=$(curl -o /dev/null -s -w "%{http_code}\n" "$url")
+
+    # Check if the status code is 200
+    if [[ $http_status -eq 200 ]]; then
```
Comment (nickboldt) [inline .ibm/pipelines/openshift-tests.sh:170]: you might also get a 302 if any redirection is happening... maybe it's better to test that you're NOT getting a 404? 

Also you could do a `curl -I` to get just the http headers, then more easily skip that for the reponse code. 

```
$➔ curl -sSI https://github.com/janus-idp/helm-backstage/releases/download/backstage-2.10.1/backstage-2.10.1.tgz | grep HTTP
HTTP/2 302 
```

Just an idea!

File: .ibm/pipelines/openshift-tests.sh [hunk 4]
```diff
@@ -153,13 +153,34 @@ run_tests() {
...
+      echo "Backstage is up and running!"
+      export BASE_URL=$url
```
Comment (nickboldt) [inline .ibm/pipelines/openshift-tests.sh:172]: you don't need to export this if you're only echoing it to the console.

File: .ibm/pipelines/openshift-tests.sh [hunk 5]
```diff
@@ -153,13 +153,34 @@ run_tests() {
...
+      echo "Attempt $i of $max_attempts: Backstage not yet available (HTTP Status: $http_status)"
+      sleep $wait_seconds
+    fi
+  done
 
-  echo "$BACKSTAGE_URL_RESPONSE"
-  export BASE_URL="https://${RELEASE_NAME}-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}"
-  echo "######## BASE URL ########"
-  echo "$BASE_URL"
+  echo "Failed to reach Backstage after $max_attempts attempts."
```
Comment (nickboldt) [inline .ibm/pipelines/openshift-tests.sh:182]: why not echo the URL that failed for easier debugging? 

```suggestion
  echo "Failed to reach Backstage at $BASE_URL after $max_attempts attempts."
```

File: .ibm/pipelines/openshift-tests.sh (hunk 4 above)
Comment (subhashkhileri) [reply .ibm/pipelines/openshift-tests.sh:172]: We are using this in our e2e-tests

**Pull request merged**